packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -0,0 +1,358 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'localizations.dart';
+import 'picker.dart';
+
+/// Default aesthetic values obtained by comparing with iOS pickers.
+const double _kItemExtent = 32.0;
+const double _kPickerWidth = 330.0;
+/// Considers setting the default background color from the theme, in the future.
+const Color _kBackgroundColor = CupertinoColors.white;
+
+
+/// The iOS date picker has its width fixed to [_kPickerWidth] in all modes.
+/// If the maximum width given to the picker is greater than [_kPickerWidth],
+/// the leftmost and rightmost column will be extended equally so that the
+/// widths match, and the picker is in the center.
+///
+/// If the maximum width given to the picker is smaller than [_kPickerWidth],
+/// the picker's layout will be broken.
+
+
+/// A countdown timer picker in iOS style.
+///
+/// This picker shows duration as hour, minute and second spinners. The duration
+/// showed has to be non negative and is limited to 23 hours 59 minutes 59 seconds.
+///
+/// Example: [12 hours |  3 min |  0 sec]
+class CupertinoCountdownTimerPicker extends StatefulWidget {
+  /// Constructs an iOS style countdown timer picker.
+  ///
+  /// [onTimerDurationChanged] is the callback when the selected duration changes
+  /// and must not be null.
+  ///
+  /// [initialTimerDuration] defaults to 0 second and is limited from 0 second
+  /// to 23 hours 59 minutes 59 seconds. Only hour, minute, and second values
+  /// are extracted from [initialTimerDuration], so specifying other fields like
+  /// millisecond or microsecond will not affect anything.
+  ///
+  /// [minuteInterval] is the granularity of the minute spinner. Must be a factor
+  /// of 60.
+  ///
+  /// [secondInterval] is the granularity of the second spinner. Must be a factor
+  /// of 60.
+  CupertinoCountdownTimerPicker({
+    this.initialTimerDuration = const Duration(),
+    this.minuteInterval = 1,
```

Comment:
Are there timers that only shows hours and minutes or minutes and seconds etc?

Reply:
on iOS, no. Or I don't know where to look for it.

Reply:
https://developer.apple.com/documentation/uikit/uidatepickermode/uidatepickermodecountdowntimer?language=objc. I would try it in a simple xcode project. Though it'd be nice to have the option regardless.